src/audio/mux/mux.c

```diff
@@ -634,9 +634,14 @@ static int mux_reset(struct comp_dev *dev)
 								  sink_list);
 			struct comp_buffer __sparse_cache *source_c = buffer_acquire(source);
 			int state = source_c->source->state;
+			int direction = source_c->source->direction;
 
 			buffer_release(source_c);
 
+			/* the state is independent of loopback-typed sources */
```

Comment:
I don't understand what that repeated comment means.

In  addition, the playback case was NOT modified by PR #6762 in this function.

Reply:
Just revised that. And we have further identified that the issue can only hit by demux on capture path, which has the sink component located on the playback path (e.g. echo reference for smart amp).